src/Avalonia.Controls/ToolTipService.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using Avalonia.Input;
+using Avalonia.Threading;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Handeles <see cref="ToolTip"/> interaction with controls.
+    /// </summary>
+    internal sealed class ToolTipService
```

Comment:
Why did you move these methods out of the `ToolTip` class itself? Just to separate concerns? I don't have a problem with that, just wondering. Also considering the fact that this class is `internal` and seems to be a singleton, why not make it a static class?

Reply:
Yes, separate concerns. I wanted to encapsulate all the control interaction logic; it may become even more complex in the future. Regarding `static`, I don't like having static state (the timer in this case). What if wanted to support multiple Dispatchers in the future? This class would have to be instantiated per UI thread. It's also more easily testable.

Reply:
👍 makes sense

Reply:
>why not make it a static class

Becaise static variables are *evil*. We even have an issue about that - https://github.com/AvaloniaUI/Avalonia/issues/1013